kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/EnvelopeEncryption.java

```diff
@@ -61,33 +62,41 @@ record KmsCacheConfig(
 
     }
 
+    public record SharedServices<K, E>(
+                                       Kms<K, E> kms,
+                                       EnvelopeEncryption.Config configuration,
+                                       DekManager<K, E> dekManager,
+                                       EncryptionDekCache<K, E> encryptionDekCache) {}
+
     @Override
-    public Config initialize(FilterFactoryContext context, Config config) throws PluginConfigurationException {
-        return config;
+    public SharedServices<K, E> initialize(FilterFactoryContext context, Config configuration) throws PluginConfigurationException {
+        Kms<K, E> kms = buildKms(context, configuration);
+
+        DekManager<K, E> dekManager = new DekManager<>(ignored -> kms, null, 5_000_000);
+        ScheduledExecutorService filterThreadExecutor = context.eventLoop();
```

Comment:
eventLoop will be `null` at Filter initialization time, maybe better to pass explicit null to encryptionDekCache

Reply:
I didn't know that. But why is it so? This thing gets initialized in the `KafkaProxyInitializer`, which is a `ChannelInitializer`, whose Javadoc says:

>  * A special {@link ChannelInboundHandler} which offers an easy way to initialize a {@link Channel} once it was
 * registered to its {@link EventLoop}.

so I'm really surprised that this can be null.

Reply:
Filter initialization happens from the `FilterChainFactory` ctor called from the main thread during startup

Reply:
Oh of course it does. Hmm.